### PR TITLE
Issue 581

### PR DESCRIPTION
**Pull request by robrobert99**
image

### CODE REVIEW

**Review by bonndan — COMMENTED** [review]
Looks very promising! :tada:

File: src/main/java/de/bonndan/nivio/model/Item.java [hunk 1]
```diff
@@ -178,15 +178,16 @@ public String getDescription() {
 
     /**
      * Adds a relation or replaces the similar one.
-     *
+     * <p>
```
[the remark this answers is not in the document]
Comment (basseur) [reply src/main/java/de/bonndan/nivio/model/Item.java:181]: For posterity:

image

File: src/main/java/de/bonndan/nivio/input/kubernetes/K8sJsonParser.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    private K8sJsonParser() {
+    }
+
+    private static K8sConfig getK8sConfig() {
+        if (k8sConfig != null) {
+            return k8sConfig;
+        } else {
+            try {
+                var testMapper = new ObjectMapper();
+                k8sConfig = testMapper.readValue(new File(FILEPATH), K8sConfig.class);
```
Comment (bonndan) [inline src/main/java/de/bonndan/nivio/input/kubernetes/K8sJsonParser.java:30]: :+1:

File: src/main/java/de/bonndan/nivio/input/kubernetes/K8sItem.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+import de.bonndan.nivio.input.kubernetes.itemadapters.ItemAdapter;
+import de.bonndan.nivio.input.kubernetes.status.Status;
+import org.springframework.lang.NonNull;
+
+import java.util.*;
+
+/**
+ * The K8sItem Class is intended to unify the K8s Objects up to an certain point and to store relation data between those.
+ */
+
```
Comment (bonndan) [inline src/main/java/de/bonndan/nivio/input/kubernetes/K8sItem.java:14]:
```suggestion
```

File: src/main/java/de/bonndan/nivio/input/kubernetes/K8sItem.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
```
Comment (bonndan) [inline src/main/java/de/bonndan/nivio/input/kubernetes/K8sItem.java:15]:
```suggestion
```

File: src/main/java/de/bonndan/nivio/assessment/kpi/KubernetesKPI.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        } else {
+            return de.bonndan.nivio.assessment.Status.RED.toString();
+        }
+    }
+
+    private SingletonMap replicaCondition(Integer replicaCount, Integer replicaCountDesired) {
+        var message = String.format("%s of %s Pods are ready", replicaCount, replicaCountDesired);
+        if (replicaCount == null) {
+            return new SingletonMap("ReadyReplicas count was null", de.bonndan.nivio.assessment.Status.ORANGE.toString());
+        } else if (replicaCountDesired == null) {
```
Comment (bonndan) [inline src/main/java/de/bonndan/nivio/assessment/kpi/KubernetesKPI.java:74]: every "else" could be removed here

File: src/main/java/de/bonndan/nivio/assessment/kpi/KubernetesKPI.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    @NonNull
+    public List<StatusValue> getStatusValues(Component component) {
+        if (!(component instanceof Labeled))
+            return new ArrayList<>();
+
+
+        var statusList = new ArrayList<StatusValue>();
+        var counter = new AtomicInteger(0);
+        ((Labeled) component).getLabels(Label.k8s).forEach((key, value) -> {
+            if (!ObjectUtils.isEmpty(value)) {
```
Comment (bonndan) [inline src/main/java/de/bonndan/nivio/assessment/kpi/KubernetesKPI.java:33]: You might inverse the if condition to reduce nesting level. Also, the if-branches might only define the "`new StatusValue(...)`" add `statusList.add(...)` could be called at the end of each iteration.

File: src/main/java/de/bonndan/nivio/input/kubernetes/K8sJsonParser.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+public class K8sJsonParser {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(K8sJsonParser.class);
+    private static final String FILEPATH = "src/main/resources/k8sLabelConfig.json";
+    private static K8sConfig k8sConfig = null;
+
+    private K8sJsonParser() {
+    }
+
```
Comment (bonndan) [inline src/main/java/de/bonndan/nivio/input/kubernetes/K8sJsonParser.java:23]: I loading the config once is sufficient:

```suggestion
static {
    config = getK8sConfig();
}
```

**Pull request merged**